src/Workspaces/Core/Portable/Workspace/Solution/Solution.cs

```diff
@@ -1442,16 +1442,45 @@ public Solution WithAnalyzerConfigDocumentTextLoader(DocumentId documentId, Text
         }
 
         /// <summary>
-        /// Creates a branch of the solution that has its compilations frozen in whatever state they are in at the time, assuming a background compiler is
-        /// busy building this compilations.
-        /// 
-        /// A compilation for the project containing the specified document id will be guaranteed to exist with at least the syntax tree for the document.
-        /// 
-        /// This not intended to be the public API, use Document.WithFrozenPartialSemantics() instead.
+        /// Creates a branch of the solution that has its compilations frozen in whatever state they are in at the time,
+        /// assuming a background compiler is busy building this compilations.
+        /// <para/> A compilation for the project containing the specified document id will be guaranteed to exist with
+        /// at least the syntax tree for the document.
+        /// <para/> This not intended to be the public API, use Document.WithFrozenPartialSemantics() instead.
         /// </summary>
         internal Solution WithFrozenPartialCompilationIncludingSpecificDocument(DocumentId documentId, CancellationToken cancellationToken)
         {
-            var newCompilationState = _compilationState.WithFrozenPartialCompilationIncludingSpecificDocument(documentId, cancellationToken);
+            // in progress solutions are disabled for some testing
+            if (this.Services.GetService<IWorkspacePartialSolutionsTestHook>()?.IsPartialSolutionDisabled == true)
+                return this;
+
+            var currentCompilationState = _compilationState;
+            var currentDocumentState = this.SolutionState.GetRequiredDocumentState(documentId);
```

Comment:
Not really critical but it does seem a bit odd that this does part of the algorithm here in Solution.cs, even though it's only operating on CompilationStates, and then calls the other method. Not bad but seems like maybe we can unify things a bit.

Reply:
yeah.  originally i was doing this entirely within SolutionCompilationState.  But it was a bit odd as i had to start with 'this' and then basically continually overwrite itself with the forked version of itself.  

let me switch back and see how i feel.

Reply:
ok.  i moved back.  lmk if you prefer it.